src/Core/Framework/Adapter/Twig/Extension/BuildBreadcrumbExtension.php

```diff
@@ -19,35 +19,28 @@ class BuildBreadcrumbExtension extends AbstractExtension
     /**
      * @internal
      *
-     * @param EntityRepository<CategoryCollection> $categoryRepository
+     * @param SalesChannelRepository<EntityCollection<SalesChannelCategoryEntity>> $categoryRepository
      */
     public function __construct(
         private readonly CategoryBreadcrumbBuilder $categoryBreadcrumbBuilder,
-        private readonly EntityRepository $categoryRepository
+        private readonly SalesChannelRepository $categoryRepository,
     ) {
     }
 
     public function getFunctions(): array
     {
         return [
-            new TwigFunction('sw_breadcrumb_full', $this->getFullBreadcrumb(...), ['needs_context' => true]),
-            new TwigFunction('sw_breadcrumb_full_by_id', $this->getFullBreadcrumbById(...), ['needs_context' => true]),
+            new TwigFunction('sw_breadcrumb_full', $this->getFullBreadcrumb(...)),
+            new TwigFunction('sw_breadcrumb_full_by_id', $this->getFullBreadcrumbById(...)),
         ];
     }
 
     /**
-     * @param array<string, mixed> $twigContext
-     *
-     * @return array<string, CategoryEntity>
+     * @return array<string, SalesChannelCategoryEntity>
      */
-    public function getFullBreadcrumb(array $twigContext, CategoryEntity $category, Context $context): array
+    public function getFullBreadcrumb(CategoryEntity $category, SalesChannelContext $context): array
```

Comment:
that is definitely a break. and we do only breaking changes until the first RC :slightly_smiling_face: 

so, you need to change it back unfortunately. but other than that, the change looks okay for me on first sight

Reply:
Imo, extending the allowed type is not break, as calls of the method would not break with this change. 

I guess the tool is just not smart enough to detect this :grin: 

so, you can add it to the ignore config

Reply:
Perfect :)